Neos.Neos/Classes/Service/LinkingService.php

```diff
@@ -305,19 +320,19 @@ public function createNodeUri(ControllerContext $controllerContext, $node = null
             $site = $this->siteRepository->findOneByNodeName($siteNodeName);
         }
 
-        $baseUri = $request->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI);
+        $baseUri = $this->baseUriProvider->getConfiguredBaseUriOrFallbackToCurrentRequest();
         if ($site->hasActiveDomains()) {
             $requestUriHost = $baseUri->getHost();
-            $activeHostPatterns = $site->getActiveDomains()->map(function ($domain) {
+            $activeHostPatterns = $site->getActiveDomains()->map(static function (Domain $domain) {
                 return $domain->getHostname();
             })->toArray();
             if (!in_array($requestUriHost, $activeHostPatterns, true)) {
                 $uri = $this->createSiteUri($controllerContext, $site) . '/' . ltrim($uri, '/');
             } elseif ($absolute === true) {
-                $uri = $baseUri. ltrim($uri, '/');
+                $uri = $baseUri . ltrim($uri, '/');
             }
         } elseif ($absolute === true) {
-            if (substr($uri, 0, 7) !== 'http://' && substr($uri, 0, 8) !== 'https://') {
+            if (strpos($uri, 'http://') !== 0 && strpos($uri, 'https://') !== 0) {
```

Comment:
nitpick, but since you changed it:
This is what the awesome "Php Inspections ​(EA Extended)​" suggest instead:
```suggestion
            if (strncmp($uri, 'http://', 7) !== 0 && strncmp($uri, 'https://', 8) !== 0) {
```